src/EscapeCodeGenerator.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "Defs.hpp"
+#include <string>
+
+struct EscapeCodeGenerator {
+
+    using EscapeCode = std::string;
+
+    virtual EscapeCode cursorForward(unsigned int) const { return EscapeCode{}; }
```

Comment:
Since you're not using auto, I believe compiler will not yell at you if you simply do `return {}`. AFAIK it was introduced in C++14 or C++11.